### PR TITLE
Add bot protection handling in http handlers

### PR DESCRIPTION
**Pull request by pkong-ds**
## What's in this PR?
- guard captcha if required in below http handlers in AuthUI
  - signup identify, all branches
  - signup-login identify, all branches
  - promote identify, all branches
  - login identify, all branches
  - login primary authentication, password, passkey (non-oob-otp branches)
  - login secondary authentication, additional password, totp, recovery code (non-oob-otp branches)
  - reauth primary authentication, password, passkey (non-oob-otp branches)
  - reauth secondary authentication, additional password, totp, recovery code (non-oob-otp branches)

### Note
This PR only implements server-side guarding (server as in SSR, not api serever)

## Upcoming
- special handling on `oob-otp-email` and `oob-otp-sms` which requires an extra screen

### CODE REVIEW

**Comment by pkong-ds** [comment]
@louischan-oursky In case you missed, this PR is ready for review 🙏

File: pkg/auth/handler/webapp/viewmodels/authflow.go [hunk 1]
```diff
@@ -43,6 +43,12 @@ type AuthflowViewModel struct {
 	LoginIDContextualType string
 
 	PasskeyRequestOptionsJSON string
+
+	PhoneLoginIDBotProtectionRequired    bool
+	EmailLoginIDBotProtectionRequired    bool
+	UsernameLoginIDBotProtectionRequired bool
+	PasskeyBotProtectionRequired         bool
+	OauthBotProtectionRequired           bool
```
Comment (louischan-oursky) [inline pkg/auth/handler/webapp/viewmodels/authflow.go:51]: OAuth

File: pkg/auth/handler/webapp/viewmodels/authflow.go [hunk 2]
```diff
@@ -63,6 +69,12 @@ func (m *AuthflowViewModeler) NewWithAuthflow(f *authflow.FlowResponse, r *http.
 	passkeyEnabled := false
 	passkeyRequestOptionsJSON := ""
 
+	bpRequiredEmail := false
+	bpRequiredPhone := false
+	bpRequiredUsername := false
+	bpRequiredPasskey := false
+	bpRequiredOauth := false
```
Comment (louischan-oursky) [inline pkg/auth/handler/webapp/viewmodels/authflow.go:76]: OAuth

File: pkg/auth/webapp/bot_protection.go [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+		}
+	}
+
+	return false, fmt.Errorf("identification type: \"%v\" not found in flow response options: [%v]", identificationType, options)
+}
+
+func IsAuthenticateStepBotProtectionRequired(authenticationType config.AuthenticationFlowAuthentication, f *authflow.FlowResponse) (bool, error) {
+	options := GetAuthenticationOptions(f)
+	for _, option := range options {
+		fmt.Printf("option: %+v\n", option)
```
Comment (louischan-oursky) [inline pkg/auth/webapp/bot_protection.go:36]: Remove this log.

**Comment by pkong-ds** [comment]
@louischan-oursky updated, thanks! Added 2 commits, starting from [Use OAuth instead of Oauth casing](https://github.com/authgear/authgear-server/pull/4475/commits/e3280ace9a126366e99e06ac77a406ae9d876132)

**Pull request merged**